app/src/main/java/com/kickstarter/viewmodels/BackingViewModel.java

```diff
@@ -275,6 +285,36 @@ public ViewModel(final @NonNull Environment environment) {
       project
         .compose(bindToLifecycle())
         .subscribe(this.koala::trackViewedPledgeInfo);
+
+      backing
+        .map(Backing::backerCompletedAt)
+        .map(ObjectUtils::isNotNull)
+        .compose(bindToLifecycle())
+        .subscribe(this.markAsReceivedIsChecked);
+
+      final Observable<Pair<Project, Backing>> projectAndBacking = Observable
+        .combineLatest(project, backing, Pair::create);
+
+      projectAndBacking
+        .compose(takePairWhen(this.markAsReceivedSwitchChecked))
+        .switchMap(triple -> this.client.postBacking(triple.first.first, triple.first.second, triple.second))
```

Comment:
Is there a way to make this line more readable? I think that using `triple`, `first` and `second` is a little bit too generic to understand what's going on.